### PR TITLE
Track past attempts from rspec-retry

### PR DESCRIPTION
**Pull request by kylekthompson**
When using rspec-retry in tandem with ABQ, this change will report any additional attempts made by rspec-retry in the `past_attempts` field.

- Track past rspec-retry attempts
- Update rspec retry snapshots
- Bump the version to 1.1.0

### CODE REVIEW

File: lib/rspec/abq/formatter.rb [hunk 1]
```diff
@@ -82,6 +87,27 @@ def self.abq_result(example)
...
+        retry_attempts.times.map do |attempt_index|
+          {
+            status: {type: :failure},
+            id: example.id,
+            display_name: example.metadata[:full_description],
+            output: retry_exceptions[attempt_index],
+            runtime: 0, # unknown
+            meta: {}
+          }
+        end
```
Comment (kylekthompson) [inline lib/rspec/abq/formatter.rb:109]: Unfortunately the ABQ protocol requires some of this because it calls past attempts just another test result spec. In the RWX Test Results schema, much of this is not necessary for a past attempt (really just the status)

Comment (kylekthompson) [reply lib/rspec/abq/formatter.rb:109]: And ultimately much of it is dropped, though the 0 duration is unfortunate

File: lib/rspec/abq/formatter.rb [hunk 2]
```diff
@@ -82,6 +87,27 @@ def self.abq_result(example)
...
+        return unless retry_attempts > 0
+
+        retry_attempts.times.map do |attempt_index|
+          {
+            status: {type: :failure},
+            id: example.id,
+            display_name: example.metadata[:full_description],
+            output: retry_exceptions[attempt_index],
+            runtime: 0, # unknown
+            meta: {}
```
Comment (ayazhafiz) [inline lib/rspec/abq/formatter.rb:107]: Is there any metadata on the rspec retry attempts? If not a comment be added that this is intentionally empty

Comment (kylekthompson) [reply lib/rspec/abq/formatter.rb:107]: Nope. I'll drop a comment in here 👍

**Review by ayazhafiz — APPROVED** [review]

File: spec/features/__snapshots__/has-consistent-output-with-rspec-retry-test-results-rspec-3.8.gemfile.snap [hunk 3]
```diff
@@ -10,7 +10,7 @@
     "otherErrors": 0,
     "pended": 3,
     "quarantined": 0,
-    "retries": 0,
```
Comment (ayazhafiz) [inline spec/features/__snapshots__/has-consistent-output-with-rspec-retry-test-results-rspec-3.8.gemfile.snap:13]: There is a bug in ABQ we should fix too:
https://github.com/rwx-research/abq/issues/497

Comment (kylekthompson) [reply spec/features/__snapshots__/has-consistent-output-with-rspec-retry-test-results-rspec-3.8.gemfile.snap:13]: Talked about this over in the ABQ issue.

TL;DR it's not a bug, it's just poorly named (my bad) https://github.com/rwx-research/test-results-schema/blob/badaa101c083d595fbf86692d7cf7ae34b0c4012/v1/summary.json#L57.
